### PR TITLE
Edited the developer installation instructions to add in missing context about repository cloning

### PR DESCRIPTION
**Pull request by cuppajoeman**
## Explanation for Changes

[Documentation Reference](https://manimce--####.org.readthedocs.build)

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [ ] My new functions/classes either have a docstring or are private
- [ ] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

File: docs/source/installation/for_dev.rst [hunk 1]
```diff
@@ -69,6 +69,9 @@ installation with Poetry.
 Installing Manim using Poetry
 *****************************
 
+#.  First start by cloning the `manim repo <https://github.com/manimcommunity/manim>`_ and moving into
+    the folder you just cloned.
```
Comment (huguesdevimeux) [inline docs/source/installation/for_dev.rst:73]: I believe it's better to write the instructions in terms of `git clone` and `cd manim`, etc., in order to be more beginner-friendly. What do you think ?

**Review by huguesdevimeux — CHANGES_REQUESTED** [review]

File: docs/source/installation/for_dev.rst (hunk 1 above)
Comment (behackl) [reply docs/source/installation/for_dev.rst]: I've changed this via ee7b3ee08f692ccad7a3c7eaecf22fa0aea4ae3f, it now links to the document where we already explain working with git.

**Review by behackl — APPROVED** [review]
I've added a link (works, tested locally) to more explicit git instructions. This is good to go from my point of view, it would be good if someone else could briefly look at this.

Thanks for your contribution!

File: docs/source/installation/for_dev.rst [hunk 2]
```diff
@@ -69,6 +69,10 @@ installation with Poetry.
 Installing Manim using Poetry
 *****************************
 
+#.  First start by cloning the `manim repo <https://github.com/manimcommunity/manim>`_ 
+    (see the first few steps in :doc:`../contributing/development` for more details),
+    then change into the freshly cloned directory.
```
Comment (Darylgolden) [inline docs/source/installation/for_dev.rst:74]:
```suggestion
    then navigate into the freshly cloned directory in the terminal.
```

File: docs/source/installation/for_dev.rst [hunk 3]
```diff
@@ -69,6 +69,10 @@ installation with Poetry.
 Installing Manim using Poetry
 *****************************
 
+#.  First start by cloning the `manim repo <https://github.com/manimcommunity/manim>`_
```
Comment (Darylgolden) [inline docs/source/installation/for_dev.rst:72]:
```suggestion
#.  First start by cloning the `manim repository <https://github.com/manimcommunity/manim>`_ 
```

**Review by Darylgolden — CHANGES_REQUESTED** [review]
Minor changes

**Review by Darylgolden — APPROVED** [review]

**Pull request merged**